src/averageAges.js

```diff
@@ -54,8 +75,26 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const allClildren = people.filter(person => {
+    return onlyWithSon
+      ? person.sex === 'm' && people.filter((mother) =>
```

Comment:
```suggestion
      ? person.sex === 'm' && people.find((mother) =>
```
Тебе надо найти только один элемент